go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -178,6 +188,34 @@ func (st *symtab) AddTable(t *table) error {
 	return nil
 }
 
+// AllTables returns an ordered list of all current tables.
+func (st *symtab) AllTables() []*table {
+	var tables []*table
```

Comment:
Minor nit, isn't this supposed to be friendlier on the GC:

```
tables := make([]*table, 0, len(st.orderedTables);
```

Reply:
Done. I also had to add a guard for length==0, which is why I went with the simpler code at the cost of GC overhead. Without the guard, the `nil` check fails.